lib/matplotlib/patches.py

```diff
@@ -137,9 +137,22 @@ def contains(self, mouseevent, radius=None):
         """
         if callable(self._contains):
             return self._contains(self, mouseevent)
+        inside = False
+        start = 0
         radius = self._process_radius(radius)
-        inside = self.get_path().contains_point(
-            (mouseevent.x, mouseevent.y), self.get_transform(), radius)
+        codes = self.get_path().codes
+        vertices = self.get_path().vertices
+        for i in range(1, len(codes)+1):
+            # fix bug (#8384)
+            # if current path is concatenated by multiple pathes
+            # check if any subpath contains the point.
+            if (i == len(codes)) or (codes[i] == 1):
+                path = Path(np.concatenate([vertices[start: i]]),
+                            np.concatenate([codes[start: i]]))
+                inside = inside or path.contains_point(
```

Comment:
once you have found that the point is in one path, I think we can short-circuit the loop and return?

Reply:
ok will do.